testsuite/MDAnalysisTests/analysis/test_persistencelength.py

```diff
@@ -30,81 +30,69 @@
 import numpy as np
 import matplotlib
 
-from numpy.testing import (
-    assert_,
-    assert_almost_equal,
-    assert_raises,
-    dec
-)
+from numpy.testing import assert_almost_equal
 
 from MDAnalysisTests.datafiles import Plength
-from MDAnalysisTests import module_not_found
 
 
-class TestPersistenceLength(TestCase):
-    def setUp(self):
-        self.u = MDAnalysis.Universe(Plength)
+class TestPersistenceLength(object):
+    @staticmethod
+    @pytest.fixture()
+    def u():
+        return MDAnalysis.Universe(Plength)
 
-    def tearDown(self):
-        del self.u
-
-    def test_ag_VE(self):
-        ags = [self.u.atoms[:10], self.u.atoms[10:110]]
-        assert_raises(ValueError, polymer.PersistenceLength, ags)
-
-    def _make_p(self):
+    @staticmethod
+    @pytest.fixture()
+    def p(u):
         ags = [r.atoms.select_atoms('name C* N*')
-               for r in self.u.residues]
+               for r in u.residues]
 
         p = polymer.PersistenceLength(ags)
         return p
 
-    def test_run(self):
-        p = self._make_p()
+    def test_ag_ValueError(self, u):
+        ags = [u.atoms[:10], u.atoms[10:110]]
+        with pytest.raises(ValueError):
+            polymer.PersistenceLength(ags)
+
+    def test_run(self, p):
         p.run()
 
-        assert_(len(p.results) == 280)
+        assert len(p.results) == 280
         assert_almost_equal(p.lb, 1.485, 3)
 
-    def test_fit(self):
-        p = self._make_p()
+    def test_fit(self, p):
         p.run()
```

Comment:
@utkbansal This `p_run` fixture should use the `p` fixture. The `p` fixture` needs to remain or `test_raise_NoDataError` will be broken.